File: Leetcode 121. Best Time to Buy and Sell Stock.js
var maxProfit = function(prices) {
    var maxPro=0;
    var minPri= prices[0] || 0;
    for (let i=0;i<prices.length;i++){
        minPri=Math.min(minPri,prices[i]);
        maxPro=Math.max(maxPro,prices[i]-minPri);
    }
    return maxPro;
};

File: Leetcode 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxpro=0;
        int minpri=INT_MAX;
        int n=prices.size();
        for(int i=0;i<n;i++){
            minpri=min(minpri,prices[i]);
            maxpro=max(maxpro,prices[i]-minpri);
        }
        return maxpro;
    }
};
